invisible_cities/core/fit_functions.py

```diff
@@ -4,16 +4,56 @@
 GML November 2016
 """
 
-import numpy  as np
-import pandas as pd
+import numpy   as np
+import pandas  as pd
+import inspect as insp
 import scipy.optimize
 import scipy.stats
 
+from functools import wraps
+
 from .                    import core_functions as coref
 from .  stat_functions    import poisson_sigma
 from .. evm.ic_containers import FitFunction
 
 
+def fixed_parameters(fn, **kwargs):
+    """
+    Wrapper to fix parameters of a function
+    which can then be used in curve_fit.
+    """
+    fn_sig      = insp.signature(fn)
+    fn_args     = np.array(list(fn_sig.parameters))[1:]
+    par_gen     = (kwargs[k] if k in kwargs.keys() else np.nan for k in fn_args)
```

Comment:
Strictly speaking arguments are values (data, objects ... the stuff you send in to a function when you call it) while parameters are names (the names which appear in the parameter list in the definiton of the function), so your choice of the variable `fn_args` for a bunch of names, is rather confusing.

Reply:
As a coding pleb I am not sensitive to these things. I'll change and update.

Reply:
We live and learn. Don't bother changing anything. I'll figure it all out eventually, and then I'll either tell you why you don't need to write it, or clean it up myself.